### PR TITLE
feature: add support for rhcc

### PR DESCRIPTION
**Pull request by wei-lee**
The change will allow us to support `rhcc` registry, and it will be the registry that will have the productized images.

To verify:

Run `ansible-playbook install-mobile-services.yml -e "ansible_playbookbundle_registry_type=rhcc"` and check the configuration of the ansible service broker. There should be a new registry added that's pointing to rhcc.

### CODE REVIEW

**Comment by psturc** [comment]
👀

File: README.md [hunk 1]
```diff
@@ -14,12 +14,29 @@ It also contains scripts for local development of Mobile Services (using `Minish
...
+    If you want to use the community releases, run the following command:
 
-```
-ansible-playbook install-mobile-services.yml
-```
+    ```
+    ansible-playbook install-mobile-services.yml
+    ```
+
+    If you want to use the productized releases from RedHat Container Catalog, please use the following command:
```
Comment (grdryn) [inline README.md]:
```suggestion
    If you want to use the productized releases from Red Hat Container Catalog, please use the following command:
```

It's always "Red Hat", never "RedHat".

File: roles/ansible-service-broker-setup/defaults/main.yml [hunk 2]
```diff
@@ -12,7 +12,11 @@ broker_mobile_configuration:
...
 #The url for the registry that hosts our APB images
+ansible_playbookbundle_registry_type: dockerhub
 ansible_playbookbundle_registry_url: https://registry.hub.docker.com
-
 #The organisation that our APB images are in
 ansible_playbook_bundle_registry_org: aerogearcatalog
+rhcc_registry_auth_name: ""
+rhcc_registry_auth_type: ""
+rhcc_registry_tag: "{{ rhcc_release_tag }}"
+rhcc_registry_url: "https://registry.access.redhat.com"
```
Comment (grdryn) [inline roles/ansible-service-broker-setup/defaults/main.yml]: I'm not sure if this should be registry.access.redhat.com or registry.redhat.io. Probably the latter, but I'll see if I can check.

Edit: I'd go with registry.redhat.io like the openshift-ansible installer uses: https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/ansible_service_broker/vars/openshift-enterprise.yml

```suggestion
rhcc_registry_url: "https://registry.redhat.io"
```

File: roles/ansible-service-broker-setup/defaults/main.yml [hunk 3]
```diff
@@ -12,7 +12,11 @@ broker_mobile_configuration:
   value: true
 
 #The url for the registry that hosts our APB images
+ansible_playbookbundle_registry_type: dockerhub
 ansible_playbookbundle_registry_url: https://registry.hub.docker.com
-
 #The organisation that our APB images are in
 ansible_playbook_bundle_registry_org: aerogearcatalog
+rhcc_registry_auth_name: ""
+rhcc_registry_auth_type: ""
```
Comment (grdryn) [inline roles/ansible-service-broker-setup/defaults/main.yml]: What should these 2 be set to? Should we default to what the openshift-ansible installer sets? https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/ansible_service_broker/tasks/install.yml#L200..L201

```suggestion
rhcc_registry_auth_type: "secret"
```

File: roles/ansible-service-broker-setup/defaults/main.yml [hunk 4]
```diff
@@ -12,7 +12,11 @@ broker_mobile_configuration:
   value: true
 
 #The url for the registry that hosts our APB images
+ansible_playbookbundle_registry_type: dockerhub
 ansible_playbookbundle_registry_url: https://registry.hub.docker.com
-
 #The organisation that our APB images are in
 ansible_playbook_bundle_registry_org: aerogearcatalog
+rhcc_registry_auth_name: ""
```
Comment (grdryn) [inline roles/ansible-service-broker-setup/defaults/main.yml]:
```suggestion
rhcc_registry_auth_name: "asb-registry-auth"
```

File: roles/ansible-service-broker-setup/defaults/main.yml (hunk 3 above)
Comment (psturc) [reply roles/ansible-service-broker-setup/defaults/main.yml]: I think it's good idea to refer to the secret created by default. Just verified that it's present on my cluster. `<openshift-url>/console/project/openshift-ansible-service-broker/browse/secrets/asb-registry-auth`

File: README.md [hunk 5]
```diff
@@ -14,12 +14,29 @@ It also contains scripts for local development of Mobile Services (using `Minish
...
+    ansible-playbook install-mobile-services.yml
+    ```
+
+    If you want to use the productized releases from Red Hat Container Catalog, please use the following command:
+
+    ```
+    ansible-playbook install-mobile-services.yml -e "ansible_playbookbundle_registry_type=rhcc"
+    ```
+
+    If authentication is required, you can set them via these variables: `rhcc_registry_auth_name` and `rhcc_registry_auth_type`.
```
Comment (psturc) [inline README.md]: We should refer to some documentation where this variables are explained, e.g. [here](https://github.com/openshift/ansible-service-broker/blob/master/docs/config.md#registry-configuration)

**Comment by wei-lee** [comment]
@grdryn @psturc all your comments are address, mind taking another look?

**Review by grdryn — APPROVED** [review]
LGTM! :100:

**Pull request merged**